File: VB/11_Images/ChangeImageSize.vb
Imports System.ComponentModel
Imports System.Text
Imports Spire.Presentation

Namespace ChangeImageSize
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub btnRun_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnRun.Click
			'Create a PPT document
			Dim presentation As New Presentation()

			'Load document from disk
			presentation.LoadFromFile("..\..\..\..\..\..\Data\ExtractImage.pptx")

			'
			Dim scale As Single=0.5f
			For Each slide As ISlide In presentation.Slides
				For Each shape As IShape In slide.Shapes
					If TypeOf shape Is IEmbedImage Then
						Dim image As IEmbedImage = TryCast(shape, IEmbedImage)
						image.Width = image.Width * scale
						image.Height = image.Height * scale
					End If
				Next shape
			Next slide

			'Save the document
			Dim result As String="ChangeImageSize_result.pptx"
			presentation.SaveToFile(result, FileFormat.Pptx2013)

			'Launch the file
			OutputViewer(result)
		End Sub
		Private Sub OutputViewer(ByVal filename As String)
			Try
				Process.Start(filename)
			Catch
			End Try
		End Sub
	End Class
End Namespace

File: CS/11_Images/ChangeImageSize.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Spire.Presentation;

namespace ChangeImageSize
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();            
        }

        private void btnRun_Click(object sender, EventArgs e)
        {
            //Create a PPT document
            Presentation presentation = new Presentation();

            //Load document from disk
            presentation.LoadFromFile(@"..\..\..\..\..\..\Data\ExtractImage.pptx");

            //
            float scale=0.5f;
            foreach (ISlide slide in presentation.Slides)
            {
                foreach (IShape shape in slide.Shapes)
                {
                    if (shape is IEmbedImage)
                    {
                        IEmbedImage image = shape as IEmbedImage;
                        image.Width = image.Width * scale;
                        image.Height = image.Height * scale;
                    }
                }
            }
            
            //Save the document
            string result="ChangeImageSize_result.pptx";
            presentation.SaveToFile(result, FileFormat.Pptx2013);

            //Launch the file
            OutputViewer(result);
        }
        private void OutputViewer(string filename)
        {
            try
            {
                System.Diagnostics.Process.Start(filename);
            }
            catch { }
        }
    }
}